curvefs/src/client/extent_manager.h

```diff
@@ -33,74 +33,113 @@
 #include "curvefs/src/client/error_code.h"
 
 using ::curvefs::metaserver::VolumeExtentList;
+using ::curvefs::metaserver::VolumeExtent;
 using ::curvefs::space::Extent;
 
 namespace curvefs {
 namespace client {
 
-
-
-/**
- * @brief  According to the extents in the file,
- * divide the [offset, len] to be written,
- * and get the extents that need to be allocated
- *
- * @param[in] extents  extents in the file
- * @param[in] offset  offset to be written
- * @param[in] len  len to be written
- * @param[out] toAllocExtents  extents that need to be allocated
- *
- * @return errcode
- */
-CURVEFS_ERROR GetToAllocExtents(const VolumeExtentList &extents,
-    uint64_t offset,
-    uint64_t len,
-    std::list<ExtentAllocInfo> *toAllocExtents);
-
-/**
- * @brief  Merge extents allocated to extents in the file
- *
- * @param[in] allocedExtents  extents allocated
- * @param[in] toAllocExtents  extents that need to be allocated
- * @param[in,out] extents  extents in the file
- *
- * @return  errcode
- */
-CURVEFS_ERROR MergeAllocedExtents(
-    const std::list<ExtentAllocInfo> &toAllocExtents,
-    const std::list<Extent> allocatedExtents,
-    VolumeExtentList *extents);
-
-/**
- * @brief mark unwritten extents to written in the file extents
- *
- * @param[in] offset  offset to be written
- * @param[in] len  len to be written
- * @param[in,out] extents  extents in the file
- *
- * @return  errcode
- */
-CURVEFS_ERROR MarkExtentsWritten(uint64_t offset, uint64_t len,
-    VolumeExtentList *extents);
-
-/**
- * @brief According to the extents in the file,
- * divide the [lOffset, len] to be read/write,
- * and get the physical extents
- *
- * @param extents  extents in the file
- * @param offset  logical offset to be read/write
- * @param len  len to be read/write
- * @param PExtent  physical extents
- *
- * @return errcode
- */
-CURVEFS_ERROR DivideExtents(const VolumeExtentList &extents,
-    uint64_t lOffset,
-    uint64_t len,
-    std::list<PExtent> *pExtents);
-
-
+class ExtentManager {
+ public:
+    ExtentManager() {}
+    virtual ~ExtentManager() {}
+
+    /**
+     * @brief  According to the extents in the file,
+     * divide the [offset, len] to be written,
+     * and get the extents that need to be allocated
+     *
+     * @param[in] extents  extents in the file
```

Comment:
@param[in] extents  currently allocated extents in the file?

Reply:
done